go/marshal/decode.go

```diff
@@ -305,6 +337,17 @@ func nomsValueDecoder(v types.Value, rv reflect.Value) {
 	rv.Set(reflect.ValueOf(v))
 }
 
+func marshalerDecoder(t reflect.Type) decoderFunc {
+	return func(v types.Value, rv reflect.Value) {
```

Comment:
I don't think needs to go in the cache. There is no computation that can be done ahead of time here.